internal/index/index.go

```diff
@@ -76,6 +84,16 @@ func New(ctx context.Context, repo Lister, ignorePacks restic.IDSet, p *restic.P
 				return nil
 			}
 
+			if res, ok := oldIdx.Packs[id]; ok {
+				if err := idx.AddPack(res.ID, size, res.Entries); err != nil {
+					return err
+				}
+				if err := oldIdx.RemovePack(id); err != nil {
```

Comment:
Why do you remove the old pack? Is this done to reduce memory usage?

Reply:
Yes that was what I was thinking. I actually don't know if the GC is really freeing something in practise, though...